billing-service/src/invoice_store.py

```
@@ -0,0 +1,30 @@
+import sqlite3
+
+
+def connect():
+    return sqlite3.connect("billing.db")
+
+
+def find_invoice(customer_name):
+    conn = connect()
+    cur = conn.cursor()
+    # Builds the WHERE clause by concatenating caller-supplied text.
+    query = "SELECT id, total FROM invoices WHERE customer = '" + customer_name + "'"
```

Ruff (0.15.21)

[error] Possible SQL injection vector through string-based query construction

(S608)